### DISCUSSION TITLE
Keep old data after validation

### DISCUSSION THREAD

**Discussion by Aghiad90** [livewire/livewire#4969] [1. Help] [answered]
Good day All , 
i just have question about what is the proper way to keep the old data of the inputs after submitting the request when the validation fail ?
to be more clear for example : 
i have two input fields (Input A : required ,wire:model="state.a", Input B : Required,wire:model="state.b" ) and button with wire:click="save"
so when people fill Input A and hit the button it will show error message that B is required but it will wipe whatever was in input A 
i know laravel has https://laravel.com/docs/8.x/requests#retrieving-old-input and i am wondering how we can implement same with livewire 

![image](https://user-images.githubusercontent.com/23388761/168676334-93c157ea-af64-4357-b445-7a88a5c4c41b.png)

**Comment by joshhanley** [comment]
@Aghiad90 the data for `input a` shouldn't be cleared, so you must have something else going on.

Can share your component class and view, so we can investigate?

Make sure to paste code as text, not as a screenshot, as we're unable to modify screenshots.

When you paste code into github, you need to surround it in triple back ticks and then specify the file type like php or blade.

For example
\```php
\\ PHP code here
\```

\```blade
\\ HTML and blade code here
\```

Thanks!

**Reply by Aghiad90** [reply]
Hello @joshhanley  , i got it , i was using regular laravel validation and i did switch it to livewire validation and now all good 
thank you for your reply

**Reply by joshhanley** [reply]
@Aghiad90 ah great, glad you got it sorted 🙂 thanks for adding an answer and marking this as answered!

**Comment by Aghiad90** [comment, marked as answer]
i was using regular laravel validation and i did switch it to livewire validation and now all good
